xbmc/cores/dvdplayer/DVDInputStreams/DVDInputStreamNavigator.cpp

```diff
@@ -471,6 +471,27 @@ int CDVDInputStreamNavigator::ProcessBlock(uint8_t* dest_buffer, int* read)
           m_iPartCount = 0;
         m_dll.dvdnav_get_position(m_dvdnav, &pos, &len);
 
+        // get chapters' timestamps if we have not cached them yet
+        if (m_mapTitleChapters.find(m_iTitle) == m_mapTitleChapters.end())
```

Comment:
are you sure that this map gets properly cleared between playback of different DVDs?